pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -802,6 +802,13 @@ func (r *ReconcileInfinispan) statefulSetForInfinispan(m *infinispanv1.Infinispa
 		controllerutil.SetControllerReference(m, pvc, r.scheme)
 		pvc.OwnerReferences[0].BlockOwnerDeletion = pointer.BoolPtr(false)
 		dep.Spec.VolumeClaimTemplates = []corev1.PersistentVolumeClaim{*pvc}
+		// Set a storage class if it specified
+		if storageClassName := m.StorageClassName(); storageClassName != "" {
+			if _, err = kube.FindStorageClass(storageClassName, r.client); err != nil {
+				return nil, err
+			}
+			pvc.Spec.StorageClassName = &storageClassName
+		}
```

Comment:
```suggestion
		// Set a storage class if it specified
		if storageClassName := m.StorageClassName(); storageClassName != "" {
			if _, err = kube.FindStorageClass(storageClassName, r.client); err != nil {
				return nil, err
			}
			pvc.Spec.StorageClassName = &storageClassName
		}
		dep.Spec.VolumeClaimTemplates = []corev1.PersistentVolumeClaim{*pvc}
```
`StorageClassName` needs to be assigned before`pvc` is copied into `dep.Spec.VolumeClaimTemplates`